src/main/java/com/blackducksoftware/integration/hub/service/NotificationService.java

```diff
@@ -171,32 +155,17 @@ public NotificationViewResults getAllNotificationViewResults(final Date startDat
         return uriResponses;
     }
 
-    private NotificationResults processNotificationsInParallel(final List<NotificationView> itemList) {
-        final SortedSet<NotificationContentItem> contentList = new TreeSet<>();
-        final List<Exception> exceptionList = new LinkedList<>();
-        NotificationResults results;
-        try (ParallelResourceProcessor<NotificationContentItem, NotificationView> parallelProcessor = createProcessor(logger)) {
-            final ParallelResourceProcessorResults<NotificationContentItem> processorResults = parallelProcessor.process(itemList);
-            contentList.addAll(processorResults.getResults());
-            exceptionList.addAll(processorResults.getExceptions());
-        } catch (final IOException ex) {
-            logger.debug("Error closing processor", ex);
-            exceptionList.add(ex);
-        } finally {
-            results = new NotificationResults(contentList, exceptionList);
-        }
+    private NotificationResults createNotificationResults(final List<CommonNotificationState> commonNotifications) throws IntegrationException {
+        final NotificationResults results;
+        final List<UriSingleResponse<? extends HubResponse>> uriResponseList = getAllLinks(commonNotifications);
+        final HubBucket bucket = hubBucketService.startTheBucket(uriResponseList);
```

Comment:
So the HubBucketService will be responsible for handling things in parallel depending on how it was instantiated?

Reply:
yes @ekerwin and I discussed this yesterday. It gives the user of the service the option to control it.

Reply:
Ok cool. Just making sure I understood completely.